pkg/cloud/tags/tags.go

```diff
@@ -79,10 +74,10 @@ func BuildParamsToTagSpecification(ec2ResourceType string, params infrav1.BuildP
 }
 
 // Ensure applies the tags if the current tags differ from the params.
-func Ensure(current infrav1.Tags, params *ApplyParams) error {
-	diff := computeDiff(current, params.BuildParams)
+func Ensure(current infrav1.Tags, params *infrav1.BuildParams, fn TagsApplyFunc) error {
```

Comment:
What if we make a builder struct instead of passing in an apply function? This way we could always extend the tags applier in the future as part of this package.

The reason I'm suggesting to go this direction is to make sure the `tags` package is still useful. In the current form, it seems very limited.

Consider something like:

```go
tags.WithEC2(..client..).Ensure/Apply()
```

Reply:
@vincepri - I like that idea, that would feel cleaner. I'll rework it now.

Reply:
@vincepri - i've made some changes to the package along the lines you mentioned.